go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -1896,20 +1926,12 @@ verifyTarget:
 
 ok:
 	tsv.requests.Add(1)
-	// If it's a begin, we should make the shutdown code
-	// wait for the call to end before it waits for tx empty.
-	if isBegin {
```

Comment:
Moving this to TxEngine introduces a race:
* Thread 1: tsv.Begin is called, which calls startRequest through execRequest.
* Thread 1: `tsv.mu` is released once startRequest returns.
* Thread 2: A call to `setServingType` is received.
* Thread 2: goes in: there are no begin requests in the wait group. So, it closes the TxEngine, or changes its state.
* Thread 1: TxEngine.Begin executes on a TxEngine, which adds to the beginWaitGroup. But the state is not the same as what it was when `startRequest` returned, potentially causing unexpected behavior.